### DISCUSSION TITLE
[GraphQL] tokenId always returning null on tokenTransferTxs

### DISCUSSION THREAD

**Discussion by MuckT** [celo-org/blockscout#915] [Q&A] [answered]
### Description

It looks like `tokenId` is no longer being correctly returned in `tokenTransferTxs`.

```graphql
{
  tokenTransferTxs(addressHash: "0xDE33e71fAECdEad20e6A8af8f362d2236CbA005f", first: 25) {
    edges {
      node {
        transactionHash
        blockNumber
        timestamp
        gasPrice
        gasUsed
        feeToken
        gatewayFee
        gatewayFeeRecipient
        input
        # Transfers associated with the TX
        tokenTransfer(first: 30) {
          edges {
            node {
              id
              fromAddressHash
              toAddressHash
              fromAccountHash
              toAccountHash
              value
              token
              tokenAddress
              tokenType
              tokenId
            }
          }
        }
      }
    }
  }
}
```

### Response

```json
{
  "node": {
    "blockNumber": 19422610,
    "feeToken": "CELO",
    "gasPrice": "25000000000",
    "gasUsed": "474849",
    "gatewayFee": "0",
    "gatewayFeeRecipient": null,
    "input": "0xf615283a000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e0000000000000000000000000000000000000000000000000000000000000000500000000000000000000000000000000000000000000000000000000000000fa00000000000000000000000000000000000000000000000000000000000001200000000000000000000000005b45daa4645f79a419811dc0657fa1b2695c6ab7000000000000000000000000000000000000000000000000000000000000018000000000000000000000000000000000000000000000000000000000000000056d75636b74000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000003f68747470733a2f2f617277656176652e6e65742f3433626f6671686c452d38457330516a544f4b76656b563239646f6678502d6e6c584f73706b7a38634f300000000000000000000000000000000000000000000000000000000000000000414c0f0d04b3e8f13e7ff2592f8f210214374393fff51b43cd7e2a4848056a30a9506968ae5f6686cffcdacc751bfabf0109f0bd99e47e241391f14834e6b6f3ce1b00000000000000000000000000000000000000000000000000000000000000",
    "timestamp": "2023-05-22T18:48:06.000000Z",
    "tokenTransfer": {
      "edges": [
        {
          "node": {
            "fromAccountHash": null,
            "fromAddressHash": "0x0000000000000000000000000000000000000000",
            "id": "Q2Vsb1RyYW5zZmVyOnsibG9nX2luZGV4Ijo4LCJ0cmFuc2FjdGlvbl9oYXNoIjoiMHg1MmUwZjQyYTQxYTFjZmZkNDc0MGM1ZDMxNTMyMDYyOWRmMzcyNTE2MGJhNTc2OTQyODZhYWQ2NjhkMTRhNGIzIn0=",
            "toAccountHash": "0xde33e71faecdead20e6a8af8f362d2236cba005f",
            "toAddressHash": "0xde33e71faecdead20e6a8af8f362d2236cba005f",
            "token": "CDN",
            "tokenAddress": "0x376f5039df4e9e9c864185d8fabad4f04a7e394a",
            "tokenId": null,
            "tokenType": "ERC-721",
            "value": null
          }
        }
      ]
    },
    "transactionHash": "0x52e0f42a41a1cffd4740c5d315320629df3725160ba57694286aad668d14a4b3"
  }
```

**Comment by arthurgousset** [comment]
This issue is still persisting as of Fri, Jun 9 11:50 (UTC).

Reproduce bug: I tested the query above using GraphiQL in the browser: https://explorer.celo.org/mainnet/graphiql

Result: `tokenId` is still `null`

<img width="850" alt="image" src="https://github.com/celo-org/blockscout/assets/46296830/53ee7b15-0be8-4510-94fa-f598223bc17d">

(Note: I had to modify the `URL` field in the UI to `https://explorer.celo.org/graphql` to run the query correctly, see screenshot)

**Comment by arthurgousset** [comment]
## Update (Mon, Jul 12)

### TLDR

+ We have identified the root cause of this issue and are working on a release.
+ We will update you when the fix is released

### More details

A simplified overview of the root cause is that:

+ `token_id` was previously a field and became an array of ids (called `token_ids`) in a recent upstream merge. 
+ `token_id` is only a valid field for ERC-721 and ERC-1155 NFTs. In all other cases (e.g. ERC20) the field is expected to be `null`.

The fix for the `tokenId` issue is included in this pull request (among other fixes):

- https://github.com/celo-org/blockscout/pull/918

**Comment by yerdua** [comment, marked as answer]
### Update (Tues, Jul 12) - resolved

The fix for missing `token_id` was deployed today. It is backward-compatible, so no changes are necessary in the queries themselves. GraphQL responses once again contain the expected `token_id` values, and this issue is considered resolved.
